### PR TITLE
DP-2807-Generate java code from schema

### PR DESCRIPTION
**Pull request by chlocarlos**

### CODE REVIEW

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import com.northconcepts.datapipeline.foundations.sourcecode.CodeWriter;
+import com.northconcepts.datapipeline.foundations.sourcecode.JavaCodeBuilder;
+
+public class GenerateJavaCodeFromSchema {
+
+    public static void main(String[] args) {
+
+        JavaCodeBuilder code = new JavaCodeBuilder();
+        CodeWriter codeWriter = code.getSourceWriter();
+        codeWriter.println("public class GenerateJavaCodeFromSchema {");
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java]: This is too [Inception](https://www.imdb.com/title/tt1375666/), let's call it `DataPipelineExample` if you have no concerns.

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        );
+
+        schema.addEntityRelationship(new EntityRelationshipDef("fk_suppliers_products")
+                .setDescription("this is fk_suppliers_products.")
+                .setPrimaryEntityName("suppliers")
+                .setForeignEntityName("products")
+                .setCardinality(RelationshipCardinality.ONE_TO_MANY)
+                .setForeignKeyFieldNames("supplier_id")
+                .setOnUpdateAction(ForeignKeyAction.CASCADE)
+                .setOnDeleteAction(ForeignKeyAction.RESTRICT));
```
Comment (DeleTaylor) [inline src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java:83]: Let's put this in a `createSchema()` method below, to not obscure the example, and replace it with:

```
SchemaDef schema = createSchema();
```

**Review by DeleTaylor — CHANGES_REQUESTED** [review]

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java (hunk 1 above)
Comment (chlocarlos) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java]: Done

File: src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java (hunk 2 above)
Comment (chlocarlos) [reply src/main/java/com/northconcepts/datapipeline/foundations/examples/schema/GenerateJavaCodeFromSchema.java:83]: Done

**Review by DeleTaylor — APPROVED** [review]

**Pull request merged**